src/printer/smt2/smt2_printer.cpp

```diff
@@ -322,30 +322,42 @@ void Smt2Printer::toStream(std::ostream& out,
     return;
   }
 
-  if (!force_nt.isNull())
+  // determine if we are printing out a type ascription, store the argument of
+  // the type ascription into type_asc_arg.
+  Node type_asc_arg;
+  if (n.getKind() == kind::APPLY_TYPE_ASCRIPTION)
   {
-    if (n.getType() != force_nt)
+    force_nt = TypeNode::fromType(
+        n.getOperator().getConst<AscriptionType>().getType());
+    type_asc_arg = n[0];
+  }
+  else if (!force_nt.isNull() && n.getType() != force_nt)
+  {
+    type_asc_arg = n;
```

Comment:
Shouldn't this be `type_asc_arg = force_nt`?

Reply:
No, force_nt is a type, type_asc_arg is a term, the latter corresponds to the "argument of the type ascription", see comments above this line.

Reply:
You are right, I misread, sorry.